jumanji/environments/routing/pacman/env.py

```diff
@@ -0,0 +1,549 @@
+            action_spec: a `specs.DiscreteArray` spec object.
+        """
+        return specs.DiscreteArray(
+            5,
+            name="action",
+        )
+
+    def __repr__(self) -> str:
+
+        return (
+            f"PacMan(\n"
+            f"\tnum_rows={self.x_size!r},\n"
+            f"\tnum_cols={self.y_size!r},\n"
+            f"\ttime_limit={self.time_limit!r}, \n"
+            f"\tgenerator={self.generator!r}, \n"
+            ")"
+        )
+
+    def reset(self, key: PRNGKey) -> Tuple[State, TimeStep[Observation]]:
+        """Resets the environment by calling the instance generator for a new instance.
+
+        Args:
+            key: A PRNGKey to use for random number generation.
+
+        Returns:
+            state: `State` object corresponding to the new state of the environment after a reset.
+            timestep: `TimeStep` object corresponding the first timestep returned by the environment
+                after a reset.
+        """
+
+        state = self.generator(key)
+
+        # Generate observation
+        obs = self._observation_from_state(state)
+
+        # Return a restart timestep of step type is FIRST.
+        timestep = restart(observation=obs)
+
+        return state, timestep
+
+    def step(
+        self, state: State, action: chex.Array
+    ) -> Tuple[State, TimeStep[Observation]]:
+        """Run one timestep of the environment's dynamics.
+
+        If an action is invalid, the agent does not move, i.e. the episode does not
+        automatically terminate.
+
+        Args:
+            state: State object containing the dynamics of the environment.
+            action: (int32) specifying which action to take: [0,1,2,3,4] correspond to
+                [Up, Right, Down, Left, No-op]. If an invalid action is taken, i.e. there is a wall
+                blocking the action, then no action (no-op) is taken.
+
+        Returns:
+            state: the new state of the environment.
+            the next timestep to be observed.
+        """
+
+        # Collect updated state based on environment dynamics
+        updated_state, collision_rewards = self._update_state(state, action)
+
+        # Create next_state from updated state
+        next_state = updated_state.replace(step_count=state.step_count + 1)  # type: ignore
+
+        # Check if episode terminates
+        num_pellets = next_state.pellets
+        dead = next_state.dead
+        time_limit_exceeded = next_state.step_count >= self.time_limit
+        all_pellets_found = num_pellets == 0
+        dead = next_state.dead == 1
+        done = time_limit_exceeded | dead | all_pellets_found
+
+        reward = jnp.asarray(collision_rewards)
+        action_mask_bool = jnp.array([True, True, True, True, False])
+        action_mask = self._compute_action_mask(next_state).astype(bool)
+        action_mask = action_mask * action_mask_bool
+
+        next_state = next_state.replace(action_mask=action_mask)  # type: ignore
```

Comment:
I don't think this needs to be in the state as you don't need the previous one to calculate the current one